test/functional/ex_cmds/oldfiles_spec.lua

```diff
@@ -1,49 +1,86 @@
-local h = require('test.functional.helpers')
+local Screen = require('test.functional.ui.screen')
+local helpers = require('test.functional.helpers')
 
-local buf     = h.curbufmeths
-local command = h.command
-local eq      = h.eq
-local execute = h.execute
-local feed    = h.feed
-local nvim    = h.nvim
+local buf, eq, execute = helpers.curbufmeths, helpers.eq, helpers.execute
+local feed, nvim, nvim_prog = helpers.feed, helpers.nvim, helpers.nvim_prog
+local set_session, spawn = helpers.set_session, helpers.spawn
 
 local shada_file = 'test.shada'
 
--- h.clear() uses "-i NONE", which is not useful for this test.
-local function clear()
+--
+-- helpers.clear() uses "-i NONE", which is not useful for this test.
+--
+local function _clear()
   if session then
     session:exit(0)
   end
-  h.set_session(h.spawn({h.nvim_prog,
-                         '-u', 'NONE',
-                         '--cmd', 'set noswapfile undodir=. directory=. viewdir=. backupdir=.',
-                         '--embed'}))
+  set_session(spawn({nvim_prog,
+                     '-u', 'NONE',
+                     '--cmd', 'set noswapfile undodir=. directory=. viewdir=. backupdir=.',
+                     '--embed'}))
 end
 
 describe(':oldfiles', function()
-  before_each(clear)
+  before_each(_clear)
 
-  it('shows most recently used files', function()
-    command('edit testfile1')
-    command('edit testfile2')
-    command('wshada ' .. shada_file)
-    command('rshada! ' .. shada_file)
-    assert(string.find(nvim('command_output', 'oldfiles'), 'testfile2'))
+  after_each(function()
     os.remove(shada_file)
   end)
+
+  local function add_padding(s)
+    return s .. string.rep(' ', 96 - string.len(s))
+  end
+
+  it('shows most recently used files', function()
+    screen = Screen.new(100, 5)
+    screen:attach()
+    execute('edit testfile1')
+    local filename1 = buf.get_name()
+    execute('edit testfile2')
+    local filename2 = buf.get_name()
+    execute('wshada ' .. shada_file)
+    execute('rshada! ' .. shada_file)
+    execute('oldfiles')
+    screen:expect([[
+      testfile2                                                                                           |
+      1: ]].. add_padding(filename1) ..[[ |
+      2: ]].. add_padding(filename2) ..[[ |
```

Comment:
This is failing on QB because of ordering:

```text
1: /home/quickbuild/buildagent/workspace/root/neovim/master/testfile2
2: /home/quickbuild/buildagent/workspace/root/neovim/master/testfile1
```

:-(